src/instrumentations/Elastic.Apm.StackExchange.Redis/ElasticApmProfiler.cs

```diff
@@ -22,29 +23,10 @@ namespace Elastic.Apm.StackExchange.Redis
 	/// </summary>
 	public class ElasticApmProfiler
 	{
-		private readonly ConcurrentDictionary<string, ProfilingSession> _executionSegmentSessions = new();
+		private readonly ConditionalWeakTable<IExecutionSegment, ProfilingSession> _executionSegmentSessions = new();
 
 		private readonly Lazy<IApmLogger> _logger;
 		private readonly Lazy<IApmAgent> _agent;
-		private static readonly Func<object, object> MessageFetcher;
```

Comment:
This is a continuation of https://github.com/elastic/apm-agent-dotnet/pull/2104 since we no longer set `db.instance` cc @stevejgordon